monitored-httpclient/src/test/java/com/github/nhenneaux/resilienthttpclient/monitoredclientpool/HttpClientPoolTest.java

```diff
@@ -141,19 +141,18 @@ void shouldUseNullTruststore() throws MalformedURLException, URISyntaxException
     }
 
     @Test
-    void shouldReturnToString() {
+    void whenCreatingAnHttpClientPoolWithAValidHostnameThenThePoolShoudBeNonEmptyAndHealthy() {
```

Comment:
I think that the name "shouldReturnToString" doesn't give your a clear idea about what the test is supposed to test, are we testing the ability of HttpClientPool to return a String ?

It started failing because :

In the previous implementation of isHealthy():
httpClientPool.getNextHttpClient().isEmpty() used to return true
when calling httpClientPool.getNextHttpClient(), in RoundRobinPool a filter is done to List<SingleIpHttpClient> with SingleIpHttpClient::isHealthy as a predicate => isHealthy returns the init value which is false, so empty.

with implementation of isHealthy():
isHealthy makes a checkHealthStatus() (because healthy.get() is false), the check is successful, returns isHealthy true, So httpClientPool.getNextHttpClient().isEmpty() don't return true in case of a valid hostname.

And I think we should stop making assertions on Strings and rather check object state.

Reply:
> are we testing the ability of HttpClientPool to return a String
A short answer: yes.

> I think we should stop making assertions on Strings and rather check object state
Any improvements are great, but they has to be considered as separate MR/PR. In the scope of this story we do not rewrite the previously written tests